examples/MvcExample/Controllers/HomeController.cs

```diff
@@ -84,34 +88,50 @@ public async Task<IActionResult> Complete([FromQuery(Name = "payment_id")]string
 
             var apiResponse = await _truelayer.Payments.GetPayment(paymentId);
 
-            IActionResult Failed(string status)
+            IActionResult Failed(string status, OneOf<PaymentMethod.BankTransfer> paymentMethod)
             {
                 ViewData["Status"] = status;
+
+                SetProviderAndSchemeId(paymentMethod);
                 return View("Failed");
             }
 
-            IActionResult Success(PaymentDetails payment)
+            IActionResult Success(PaymentDetails payment, OneOf<PaymentMethod.BankTransfer> paymentMethod)
             {
                 ViewData["Status"] = payment.Status;
+                SetProviderAndSchemeId(paymentMethod);
                 return View("Success");
             }
 
-            IActionResult Pending(PaymentDetails payment)
+            IActionResult Pending(PaymentDetails payment, OneOf<PaymentMethod.BankTransfer> paymentMethod)
             {
                 ViewData["Status"] = payment.Status;
+                SetProviderAndSchemeId(paymentMethod);
                 return View("Success");
             }
 
+            void SetProviderAndSchemeId(OneOf<PaymentMethod.BankTransfer> paymentMethod)
+            {
+                (string providerId, string schemeId) = paymentMethod.Match(
+                    bankTransfer => bankTransfer.ProviderSelection.Match(
+                        userSelected => (userSelected.ProviderId, userSelected.SchemeId),
+                        preselected => (preselected.ProviderId, preselected.SchemeId)
+                    ));
+
+                ViewData["ProviderId"] = providerId;
+                ViewData["SchemeId"] = schemeId;
+            }
+
             if (!apiResponse.IsSuccessful)
-                return Failed(apiResponse.StatusCode.ToString());
+                return Failed(apiResponse.StatusCode.ToString(), null!);
```

Comment:
Won't this `null!` throw an exception when it gets into `SetProviderAndSchemeId`? (`paymentMethod.Match...`)

Reply:
done in [1410b2e](https://github.com/TrueLayer/truelayer-dotnet/pull/136/commits/1410b2e64183058c017711716c75675470c9ff27)
Thank you